### PR TITLE
FIX: Display View/Edit cosigner screen for electrum-json and enable replacing electrum seed with xpub and vice versa

### PR DESCRIPTION
**Pull request by ojokne**
This PR fixes the issue with View/Edit Cosigner screen if you imported Electrum wallet file, the screen would hang on loading screen

it also makes it possible to forget electrum seed and use xpub and to import the seed back

Screenshots of View/Edit Cosigner screen
Screenshot_1787378426&nbsp;&nbsp;Screenshot_1787378468

To reproduce the issue
import any electrum wallet file([sample wallet file](https://github.com/BlueWallet/BlueWallet/blob/9651f7c49706bb345601cdec52fb723b45d46747/tests/unit/fixtures/electrum-multisig-wallet-with-seed.json)), then go `manage keys`

### CODE REVIEW

File: class/wallets/multisig-hd-wallet.ts [hunk 1]
```diff
@@ -1214,8 +1239,15 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
     const index = externalIndex - 1;
     const fingerprint = this._cosignersFingerprints[index];
     if (!MultisigHDWallet.isXpubValid(this._cosigners[index])) throw new Error('This cosigner doesnt contain valid xpub');
-    if (!bip39.validateMnemonic(mnemonic)) throw new Error('Not a valid mnemonic phrase');
-    if (fingerprint !== MultisigHDWallet.mnemonicToFingerprint(mnemonic, passphrase)) {
+    if (!bip39.validateMnemonic(mnemonic) && !MultisigHDWallet.isValidElectrumSeed(mnemonic))
+      throw new Error('Not a valid mnemonic phrase');
+    let calculatedFingerprint: string | undefined;
+    if (MultisigHDWallet.isValidElectrumSeed(mnemonic)) {
+      mnemonic = ELECTRUM_SEED_PREFIX + mnemonic;
```
Comment (GladosBlueWallet) [inline class/wallets/multisig-hd-wallet.ts:1243]: **[HIGH]** Blind prefix append. Paste `electrumseed:…` from an export and you get `electrumseed:electrumseed:…` — fingerprint mismatch or invalid seed. The enrichment center accepts malformed input; your cosigner replacement should not.

File: tests/unit/multisig-hd-wallet.test.js [hunk 2]
```diff
@@ -757,6 +757,16 @@ describe('multisig-wallet (native segwit)', () => {
     w.addCosigner(process.env.MNEMONICS_COBO);
     assert.strictEqual(w.getFingerprint(1), 'DEADBABE');
     assert.strictEqual(w.getCosignerForFingerprint('DEADBABE'), process.env.MNEMONICS_COLDCARD);
+
+    // isValidElectrumSeed method can correctly detect electrum seed
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js]: **[HIGH]** Two boolean asserts on `isValidElectrumSeed` — that's testing the validator, not the experiment. Zero coverage for getSecret→setSecret round-trip (the hang fix), electrum `replaceCosignerXpubWithSeed`, or prefixed-seed paste. CONTRIBUTING.md explicitly forbids this flavor of decorative testing.

File: class/wallets/multisig-hd-wallet.ts [hunk 3]
```diff
@@ -591,12 +608,20 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
 
         default:
           if (key && value && MultisigHDWallet.isXpubString(value.trim())) {
-            this.addCosigner(value.trim(), key, customPathForCurrentCosigner);
+            const path = customPathForCurrentCosigner ?? this._derivationPath;
+            this.addCosigner(value.trim(), key, path);
           } else if (key.replace('#', '').trim() === 'derivation') {
             customPathForCurrentCosigner = value.trim();
           } else if (key === 'seed') {
```
Comment (GladosBlueWallet) [inline class/wallets/multisig-hd-wallet.ts:612]: **[HIGH]** The hang fix lives here — electrum seed re-import with fingerprint and path fallback — yet no test proves `setSecret(wallet.getSecret())` survives an electrum-json wallet. You're shipping the cure without verifying the subject didn't die on the table.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Subject matter: electrum-json multisig cosigner handling — fix infinite loader on View/Edit (getSecret→setSecret round-trip) plus electrum seed cosigner replacement. Core wallet logic is mostly coherent; I found one real paste-path bug and test coverage that wouldn't pass a potato.

File: class/wallets/multisig-hd-wallet.ts (hunk 3 above)
Comment (ojokne) [reply class/wallets/multisig-hd-wallet.ts:612]: I have added test cases

File: tests/unit/multisig-hd-wallet.test.js (hunk 2 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js]: tests added from line 1844 to 1878

File: class/wallets/multisig-hd-wallet.ts (hunk 1 above)
Comment (ojokne) [reply class/wallets/multisig-hd-wallet.ts:1243]: I have modified to only add the prefix only if its not there, added a  unit test case for such a scenario

File: screen/wallets/ViewEditMultisigCosigners.tsx [hunk 4]
```diff
@@ -397,7 +397,8 @@ const ViewEditMultisigCosigners: React.FC = () => {
 
       const hd = new HDSegwitBech32Wallet();
       hd.setSecret(mnemonic);
-      if (!hd.validateMnemonic()) return presentAlert({ message: loc.multisig.invalid_mnemonics });
+      if (!hd.validateMnemonic() && !MultisigHDWallet.isValidElectrumSeed(mnemonic))
+        return presentAlert({ message: loc.multisig.invalid_mnemonics });
       try {
         wallet.replaceCosignerXpubWithSeed(cosignerNum, hd.getSecret(), passphrase);
```
Comment (GladosBlueWallet) [inline screen/wallets/ViewEditMultisigCosigners.tsx:422]: **[HIGH]** You validate the raw mnemonic, then feed `hd.getSecret()` into replace — `setSecret()` strips colons and turns `electrumseed:word…` into an extra word. Green light on input, rejection at replace. Paste-from-export users get a lovely error cake.

File: tests/unit/multisig-hd-wallet.test.js [hunk 5]
```diff
@@ -1822,6 +1846,36 @@ describe('multisig-wallet (native segwit)', () => {
     assert.strictEqual(w.getFingerprint(1), fp1cobo);
     assert.strictEqual(w.getCustomDerivationPathForCosigner(1), path);
     assert.strictEqual(w.getCosignerPassphrase(1), undefined);
+
+    const electrumWallet = new MultisigHDWallet();
+    const electrumJson = require('./fixtures/electrum-multisig-wallet-with-seed.json');
+
+    electrumWallet.setSecret(JSON.stringify(electrumJson));
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js:1856]: **[HIGH]** Your FIX commit exists because `getSecret()` → `setSecret()` threw on Electrum seeds and froze the screen. Zero roundtrip test for that path. Regression bait — future you will love debugging infinite loading again.

File: tests/unit/multisig-hd-wallet.test.js [hunk 6]
```diff
@@ -1822,6 +1846,36 @@ describe('multisig-wallet (native segwit)', () => {
...
+    assert.strictEqual(
+      electrumWallet.getCosigner(1),
+      'electrumseed:during pride layer jelly admit army want melody check witness favorite prosper',
+    );
+    electrumWallet.replaceCosignerSeedWithXpub(1);
+    assert.strictEqual(
+      electrumWallet.getCosigner(1),
+      'Zpub6yUED4tEZDX9v4RbbrJVfEMSHJu7yFGfpbHt4jxf48m5oEFGWCXtu32o1wQkEbCCrHJfRbc8GeoBwpRowcvTMHruNcsbm97QD4uUzaXrtNK',
+    );
+    electrumWallet.replaceCosignerXpubWithSeed(1, 'during pride layer jelly admit army want melody check witness favorite prosper');
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js]: **[HIGH]** Electrum replace tests hit `MultisigHDWallet` directly and never assert fingerprint preservation — BIP39 paths do at line 1779. You proved string storage, not that the cosigner still matches the xpub. Mock-adjacent theater.

File: tests/unit/multisig-hd-wallet.test.js [hunk 7]
```diff
@@ -757,6 +757,30 @@ describe('multisig-wallet (native segwit)', () => {
     w.addCosigner(process.env.MNEMONICS_COBO);
     assert.strictEqual(w.getFingerprint(1), 'DEADBABE');
     assert.strictEqual(w.getCosignerForFingerprint('DEADBABE'), process.env.MNEMONICS_COLDCARD);
+
+    // isValidElectrumSeed method can correctly detect electrum seed
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js]: **[HIGH]** `isValidElectrumSeed` checks live inside `basic operations work` like spare parts in a storage cube. No negative passphrase case, no UI-path test through `HDSegwitBech32Wallet`. Edge cases? This facility doesn't do those.

File: class/wallets/multisig-hd-wallet.ts [hunk 8]
```diff
@@ -1214,8 +1239,17 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
...
+      throw new Error('Not a valid mnemonic phrase');
+    let calculatedFingerprint: string | undefined;
+    if (MultisigHDWallet.isValidElectrumSeed(mnemonic)) {
+      if (!mnemonic.startsWith(ELECTRUM_SEED_PREFIX)) {
+        mnemonic = ELECTRUM_SEED_PREFIX + mnemonic;
+      }
+      const electrumSeed = MultisigHDWallet.convertElectrumMnemonicToSeed(mnemonic, passphrase);
+      calculatedFingerprint = MultisigHDWallet.seedToFingerprint(electrumSeed);
+    }
+    if (fingerprint !== MultisigHDWallet.mnemonicToFingerprint(mnemonic, passphrase) && fingerprint !== calculatedFingerprint) {
```
Comment (GladosBlueWallet) [inline class/wallets/multisig-hd-wallet.ts]: **[MEDIUM]** Electrum branch already set `calculatedFingerprint`, yet you still run BIP39 `mnemonicToFingerprint` on a prefixed seed string. Wrong algorithm, useless work — maintenance hazard for the next test subject.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**Intent:** Bug fix plus feature — unstick View/Edit co-signers on Electrum JSON import, and allow swapping an xpub cosigner for an Electrum seed.

**Verdict:** Core wallet logic is mostly coherent, but the UI still validates one string and replaces with another after BIP39 normalization mangles `electrumseed:` input, and the test chamber never reproduces the loading-screen roundtrip this PR claims to fix. Incomplete coverage; not cake.

File: screen/wallets/ViewEditMultisigCosigners.tsx (hunk 4 above)
Comment (ojokne) [reply screen/wallets/ViewEditMultisigCosigners.tsx:422]: fixed, now passing `hd.getSecret()` not `mnemonic`

File: tests/unit/multisig-hd-wallet.test.js (hunk 6 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js]: test for fingerprint added

File: class/wallets/multisig-hd-wallet.ts (hunk 8 above)
Comment (ojokne) [reply class/wallets/multisig-hd-wallet.ts]: fixed

File: tests/unit/multisig-hd-wallet.test.js [hunk 9]
```diff
@@ -1822,6 +1846,39 @@ describe('multisig-wallet (native segwit)', () => {
     assert.strictEqual(w.getFingerprint(1), fp1cobo);
     assert.strictEqual(w.getCustomDerivationPathForCosigner(1), path);
     assert.strictEqual(w.getCosignerPassphrase(1), undefined);
+
+    const electrumWallet = new MultisigHDWallet();
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js:1853]: **[HIGH]** You fixed a getSecret→setSecret throw that froze the co-signer screen, yet no test round-trips an Electrum-seed multisig through getSecret/setSecret. Regression bait. Even I wouldn't call that science.

File: tests/unit/multisig-hd-wallet.test.js (hunk 7 above)
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js]: **[HIGH]** isValidElectrumSeed assertions are grafted onto the 100-line 'basic operations work' corpse instead of their own case. CONTRIBUTING says don't bloat tests — this chamber rejects bloated test subjects.

File: tests/unit/multisig-hd-wallet.test.js [hunk 10]
```diff
@@ -1822,6 +1846,39 @@ describe('multisig-wallet (native segwit)', () => {
...
+      'electrumseed:during pride layer jelly admit army want melody check witness favorite prosper',
+    );
+    assert.strictEqual(electrumWallet.getFingerprint(1), '8AAA5D05');
+    electrumWallet.replaceCosignerSeedWithXpub(1);
+    assert.strictEqual(
+      electrumWallet.getCosigner(1),
+      'Zpub6yUED4tEZDX9v4RbbrJVfEMSHJu7yFGfpbHt4jxf48m5oEFGWCXtu32o1wQkEbCCrHJfRbc8GeoBwpRowcvTMHruNcsbm97QD4uUzaXrtNK',
+    );
+    assert.strictEqual(electrumWallet.getFingerprint(1), '8AAA5D05');
+    electrumWallet.replaceCosignerXpubWithSeed(1, 'during pride layer jelly admit army want melody check witness favorite prosper');
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js]: **[HIGH]** replaceCosignerXpubWithSeed happy-path only — no Electrum seed + passphrase edge case despite electrum-multisig-wallet-with-seed-and-passphrase.json sitting right there. Edge cases aren't optional. They're the part that explodes.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Bugfix/feature PR: Electrum-seed cosigners in multisig vaults — fixes the View/Edit co-signers infinite spinner after Electrum JSON import, enables xpub↔Electrum-seed replacement. Core wallet logic looks coherent; test coverage has conspicuous holes where the hang actually lived.

### Inline findings (could not anchor on diff)

- `screen/wallets/ViewEditMultisigCosigners.tsx:163` — **[MEDIUM]** setSecret still runs naked in useFocusEffect — no try/catch, so any future parse throw leaves isLoading true forever. You patched one throwable; the spinner trap door remains.

File: tests/unit/multisig-hd-wallet.test.js (hunk 10 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js]: updated the test case to use electrum seed that has passphrase

File: tests/unit/multisig-hd-wallet.test.js (hunk 7 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js]: Moved the `isValidElectrumSeed` tests to it own case

File: tests/unit/multisig-hd-wallet.test.js (hunk 9 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js:1853]: the `setSecret` function is actually used in line 1853

File: tests/unit/multisig-hd-wallet.test.js [hunk 11]
```diff
@@ -1822,6 +1847,43 @@ describe('multisig-wallet (native segwit)', () => {
     assert.strictEqual(w.getFingerprint(1), fp1cobo);
     assert.strictEqual(w.getCustomDerivationPathForCosigner(1), path);
     assert.strictEqual(w.getCosignerPassphrase(1), undefined);
+
+    const electrumWallet = new MultisigHDWallet();
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js:1853]: **[HIGH]** You fixed a `getSecret()`→`setSecret()` hang, then never tested that round-trip on an Electrum-json wallet with seed+passphrase—the exact path `ViewEditMultisigCosigners` clones. Congratulations: your regression test is missing. Again.

Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js:1853]: **[HIGH]** Thirty-five lines of unrelated Electrum assertions welded onto a 135-line `can edit cosigners` test. CONTRIBUTING asked for lean tests; you built a katamari. Extract it.

File: screen/wallets/ViewEditMultisigCosigners.tsx [hunk 12]
```diff
@@ -160,7 +160,9 @@ const ViewEditMultisigCosigners: React.FC = () => {
             w.current = new MultisigHDWallet();
             w.current.setNativeSegwit();
           } else {
-            tempWallet.current.setSecret(w.current.getSecret());
+            try {
+              tempWallet.current.setSecret(w.current.getSecret());
+            } catch (_) {}
```
Comment (GladosBlueWallet) [inline screen/wallets/ViewEditMultisigCosigners.tsx]: **[MEDIUM]** Empty `catch` here—setSecret fails, spinner clears, subject gets an empty or half-parsed vault clone, zero alerts. You traded infinite loading for silent data amnesia. Charming.

File: screen/wallets/ViewEditMultisigCosigners.tsx [hunk 13]
```diff
@@ -160,7 +160,9 @@ const ViewEditMultisigCosigners: React.FC = () => {
             w.current = new MultisigHDWallet();
             w.current.setNativeSegwit();
           } else {
-            tempWallet.current.setSecret(w.current.getSecret());
+            try {
+              tempWallet.current.setSecret(w.current.getSecret());
```
Comment (GladosBlueWallet) [inline screen/wallets/ViewEditMultisigCosigners.tsx:164]: **[MEDIUM]** You reuse `tempWallet` without resetting before `setSecret`. Mid-parse throw leaves partial cosigners in the ref; the UI renders Frankenstein's vault. Fresh instance first, then parse—or enjoy the corpse.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Bugfix PR: Electrum-json multisig vaults hang on View/Edit Co-signers because `getSecret()`→`setSecret()` chokes on `electrumseed:` colons and Electrum fingerprints; also adds Electrum seed replace/edit. Core wallet parsing looks coherent. Tests skip the exact round-trip that caused the hang, and the UI swallows clone failures like a black hole.

File: screen/wallets/ViewEditMultisigCosigners.tsx (hunk 13 above)
Comment (ojokne) [reply screen/wallets/ViewEditMultisigCosigners.tsx:164]: updated the try catch block

File: screen/wallets/ViewEditMultisigCosigners.tsx (hunk 12 above)
Comment (ojokne) [reply screen/wallets/ViewEditMultisigCosigners.tsx]: added an alert telling user cosigner data is invalid

File: tests/unit/multisig-hd-wallet.test.js (hunk 11 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js:1853]: extract the test to its own case `can replace electrum seed with xpub and can replace xpub with electrum seed`

File: screen/wallets/ViewEditMultisigCosigners.tsx [hunk 14]
```diff
@@ -160,10 +160,14 @@ const ViewEditMultisigCosigners: React.FC = () => {
...
-            if (!cancelled) {
-              setWalletData(new Array(tempWallet.current.getN()));
-              setWallet(tempWallet.current);
+            try {
+              tempWallet.current.setSecret(w.current.getSecret());
+              if (!cancelled) {
+                setWalletData(new Array(tempWallet.current.getN()));
+                setWallet(tempWallet.current);
+              }
+            } catch (_) {
```
Comment (GladosBlueWallet) [inline screen/wallets/ViewEditMultisigCosigners.tsx:169]: **[MEDIUM]** Catch swallows the throw, sets `hasLoaded=true`, clears loading, never sets `wallet`. Subject gets an empty cosigner list and a generic 'invalid co-signer' alert. Infinite spinner fixed; infinite confusion installed.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**Intent:** FIX — Electrum-JSON multisig vaults hung View/Edit Cosigners because `getSecret()` → `setSecret()` round-trip choked on `electrumseed:` txt lines; also ADD electrum-seed ↔ xpub cosigner replacement.

**Verdict:** Core wallet logic looks sound. Two gaps: the exact hang regression path has no test, and the UI catch trades infinite spinner for a blank broken screen. Not cake-worthy. Yet.

### Inline findings (could not anchor on diff)

- `tests/unit/multisig-hd-wallet.test.js:1471` — **[HIGH]** You fixed the hang at `getSecret()` → `setSecret()` for `electrumseed:` txt lines, then forgot to test it. The replace test at 1852 never touches that path. CONTRIBUTING wants happy-path coverage; this is the happy path. Regression bait.

File: screen/wallets/ViewEditMultisigCosigners.tsx (hunk 14 above)
Comment (ojokne) [reply screen/wallets/ViewEditMultisigCosigners.tsx:169]: I have added an alert  in the catch block and when user clicks `Ok` they are taken back to the previous screen (wallet transactions scren)

File: tests/unit/multisig-hd-wallet.test.js (hunk 11 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js:1853]: Added a test to make sure `getSecret` returns what's expected after setting a secret using `setSecret`

File: tests/unit/multisig-hd-wallet.test.js [hunk 15]
```diff
@@ -1824,6 +1849,64 @@ describe('multisig-wallet (native segwit)', () => {
...
+      "# derivation: m/1'\n" +
+      'seed: electrumseed:diagram grape account sustain bright member ethics strategy burger senior capital enforce - BlueWallet\n' +
+      '# warning! sensitive information, do not disclose ^^^ \n' +
+      '\n' +
+      "# derivation: m/48'/0'/0'/2'\n" +
+      'seed: ZprvAqPkyb5ridHr1gGiqSuAWcsrZ6jqv31Vyuaj4fzVcgt8v6PXH9tSigE6F8iw8pL16HWnhzEsXvJ5ur9HKvkAW16oHZuFeEYA1CBdsGGDFFB\n' +
+      '# warning! sensitive information, do not disclose ^^^ \n' +
+      '\n';
+
+    assert.strictEqual(electrumWallet.getSecret(), expected);
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js]: **[HIGH]** You fixed the re-import hang, then tested a postcard (`getSecret()` string equality) instead of the actual experiment: `new MultisigHDWallet().setSecret(electrumWallet.getSecret())`. ViewEditMultisigCosigners lives on that round-trip. Congratulations — untested primary hypothesis.

File: tests/unit/multisig-hd-wallet.test.js [hunk 16]
```diff
@@ -1824,6 +1849,64 @@ describe('multisig-wallet (native segwit)', () => {
...
+      'diagram grape account sustain bright member ethics strategy burger senior capital enforce',
+      'BlueWallet',
+    );
+    assert.strictEqual(
+      electrumWallet.getCosigner(1),
+      'electrumseed:diagram grape account sustain bright member ethics strategy burger senior capital enforce',
+    );
+    assert.strictEqual(electrumWallet.getFingerprint(1), '8DE7B2C3');
+    electrumWallet.replaceCosignerSeedWithXpub(1);
+    assert.throws(
```
Comment (GladosBlueWallet) [inline tests/unit/multisig-hd-wallet.test.js:1886]: **[HIGH]** One `assert.throws` for prefixed input, zero coverage for wrong passphrase or fingerprint mismatch on Electrum replace. Edge cases exist; your test protocol pretends they don't. Science requires failure modes, not just applause.

File: screen/wallets/ViewEditMultisigCosigners.tsx [hunk 17]
```diff
@@ -160,10 +160,27 @@ const ViewEditMultisigCosigners: React.FC = () => {
...
```
Comment (GladosBlueWallet) [inline screen/wallets/ViewEditMultisigCosigners.tsx:169]: **[MEDIUM]** Catch returns early — `hasLoaded` never flips, `setIsLoading(false)` never runs. Test subjects get an infinite spinner under the error dialog until they press OK and flee. Elegant, in the way a broken portal gun is elegant.

Comment (GladosBlueWallet) [inline screen/wallets/ViewEditMultisigCosigners.tsx:169]: **[MEDIUM]** Blanket `catch (_)` labels every `setSecret` failure 'invalid co-signer' — parser bugs, OOM, duplicate fingerprints, all the same apology. Misdiagnosis is not recovery; it's how you lose the plot.

File: screen/wallets/ViewEditMultisigCosigners.tsx [hunk 18]
```diff
@@ -397,7 +414,8 @@ const ViewEditMultisigCosigners: React.FC = () => {
 
       const hd = new HDSegwitBech32Wallet();
       hd.setSecret(mnemonic);
```
Comment (GladosBlueWallet) [inline screen/wallets/ViewEditMultisigCosigners.tsx:418]: **[MEDIUM]** Routing Electrum seeds through `HDSegwitBech32Wallet.setSecret()` applies BIP39 word-completion — wrong algorithm, wrong ward. Full words usually survive; partial entry gets 'helpfully' corrupted before validation. Pick a validator that doesn't mutilate the specimen first.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**Bugfix/feature PR** — Electrum-json multisig vaults were bricking View/Edit Co-signers because `getSecret()`→`setSecret()` round-trip couldn't parse `electrumseed:` lines; this adds Electrum seed validation, txt-parser fixes (colon split, path fallback, fingerprint derivation), and xpub↔Electrum seed replacement. Core wallet logic looks targeted. **Test chamber gap:** the hang-fix path (`setSecret(getSecret())`) is never exercised — only a `getSecret()` string snapshot. Error handler leaves the loading spinner running under the alert.

File: screen/wallets/ViewEditMultisigCosigners.tsx (hunk 18 above)
Comment (ojokne) [reply screen/wallets/ViewEditMultisigCosigners.tsx:418]: Not part of the changes

File: tests/unit/multisig-hd-wallet.test.js (hunk 15 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js]: Added test case for `new MultisigHDWallet().setSecret(electrumWallet.getSecret())`

File: tests/unit/multisig-hd-wallet.test.js (hunk 16 above)
Comment (ojokne) [reply tests/unit/multisig-hd-wallet.test.js:1886]: Added test case for wrong passphrase

File: screen/wallets/ViewEditMultisigCosigners.tsx (hunk 17 above)
Comment (ojokne) [reply screen/wallets/ViewEditMultisigCosigners.tsx:169]: added `setIsloading(false)` and `hasLoaded.current = true;` to catch block

**Review by GladosBlueWallet — APPROVED** [review]
**Test subject 8863:** Electrum-JSON multisig vaults were freezing the View/Edit co-signers chamber — `getSecret()`→`setSecret()` could not survive an `electrumseed:` round-trip. The fix teaches the parser colon-safe splitting, derivation-path fallback, and Electrum fingerprint derivation so those lines stop disintegrating on re-entry.

**Secondary experiment:** xpub↔Electrum-seed replacement via `isValidElectrumSeed`, refreshed replace helpers, UI validation with try-catch, and unit tests — including the round-trip path. Delightful. Actual coverage.

**Verdict:** Behavior matches the hypothesis. Commits wear proper CONTRIBUTING prefixes. No anomalies detected in this chamber. Proceed — before I find something else to optimize.

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by Overtorment — APPROVED** [review]
lgtm

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Review by ncoelho — APPROVED** [review]
